### ISSUE TITLE
Val metrics degrate during multi-GPU training

### ISSUE THREAD

**Issue by apatsekin** [keras-team/keras#13752]
<em>Please make sure that this is a Bug or a Feature Request and provide all applicable information asked by the template.
If your issue is an **implementation question**, please ask your question on [StackOverflow](http://stackoverflow.com/questions/tagged/keras) or [on the Keras Slack channel](https://keras-slack-autojoin.herokuapp.com/) instead of opening a GitHub issue.</em>  

**System information**  
- Have I written custom code (as opposed to using example directory):  
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):  Docker nvcr.io/nvidia/tensorflow:19.10-py3
- TensorFlow backend (yes / no):  yes
- TensorFlow version:  1.14.0
- Keras version:  2.3.1
- Python version:  3.6.8
- CUDA/cuDNN version:  10.2
- GPU model and memory:  8x RTX 2080 Ti
- **Using tf.keras**

**Describe the current behavior**  
While training with Horovod or internal multi_gpu_model I stumble upon very similar training behaivor, when loss/acc of training data goes fine, however validation metrics steeply drop after 1-2 epochs. Provided logs come from 8 GPU training.
**Describe the expected behavior**  
ML description of this behaivor would be "overfitting", but drop of val_acc from 0.91 to 0.04 on 10 classes is not overfitting, but seems like some kind of bug. With single GPU training everything works fine.

**Code to reproduce the issue**  
```
model.fit_generator(
        generator=train_generator,
        epochs=config.epochs,
        steps_per_epoch=(config.train_num // config.batch_size) // (hvd.size(),
        validation_data=val_generator,
        #for validation we oversample things twice per node
        validation_steps=(config.val_num // config.batch_size)  // (hvd.size()//2),
        verbose=1 if hvd.rank() == 0 else 0,
        use_multiprocessing = True,
        workers = 15,
        callbacks=callbacks_)
```
^ so the validation dataset is 1/4 of original one for each GPU, random=True. One of the callbacks suppose to average those metrics between 8 GPUs.

Horovod callbacks:
`callbacks.append(hvd.callbacks.BroadcastGlobalVariablesCallback(0))        callbacks.append(hvd.callbacks.MetricAverageCallback())`

Optimizer:
`optimizers.Adam, beta_1=0.9, beta_2=0.999, amsgrad=False`

There is also LR schedule callback that scales LR from 0.001 to 0.008 during first 4 epochs (as a warm up for multi-GPU training).

**Other info / logs**  
```
6558/6559 [============================>.] - ETA: 0s - loss: 0.0283 - acc: 0.8943   ^@^@@
Epoch 00001: val_acc improved from -inf to 0.91474, saving model to 01-0.02-0.9147.hdf5
6559/6559 [==============================] - 3557s 542ms/step - loss: 0.0283 - acc: 0.8951 - val_loss: 0.0232 - val_acc: 0.9147

Epoch 00002: LearningRateScheduler reducing learning rate to 0.003333333380830785.
Epoch 2/100
6558/6559 [============================>.] - ETA: 0s - loss: 0.0263 - acc: 0.9023
Epoch 00002: val_acc did not improve from 0.91474
6559/6559 [==============================] - 3440s 524ms/step - loss: 0.0263 - acc: 0.8999 - val_loss: 0.9915 - val_acc: 0.4873

Epoch 00003: LearningRateScheduler reducing learning rate to 0.0056666667473812896.
Epoch 3/100
6558/6559 [============================>.] - ETA: 0s - loss: 0.0238 - acc: 0.9060
Epoch 00003: val_acc did not improve from 0.91474
6559/6559 [==============================] - 3441s 525ms/step - loss: 0.0238 - acc: 0.9070 - val_loss: 2.7195 - val_acc: 0.3251

Epoch 00004: LearningRateScheduler reducing learning rate to 0.007999999997516473.
Epoch 4/100
6558/6559 [============================>.] - ETA: 0s - loss: 0.0236 - acc: 0.9064
Epoch 00004: val_acc did not improve from 0.91474
6559/6559 [==============================] - 3416s 521ms/step - loss: 0.0235 - acc: 0.9076 - val_loss: 3.8553 - val_acc: 0.0432

Epoch 00005: LearningRateScheduler reducing learning rate to 0.00800000037997961.
Epoch 5/100
6558/6559 [============================>.] - ETA: 0s - loss: 0.0206 - acc: 0.9160
Epoch 00005: val_acc did not improve from 0.91474
6559/6559 [==============================] - 3396s 518ms/step - loss: 0.0206 - acc: 0.9159 - val_loss: 2.7199 - val_acc: 0.3250
```

One more example with shorter epochs (check val_acc):
```
1639/1639 [==============================] - 1140s 696ms/step - loss: 0.0413 - acc: 0.8550 - val_loss: 0.0310 - val_acc: 0.8878

Epoch 00002: LearningRateScheduler reducing learning rate to 0.003333333380830785.
Epoch 2/100
1638/1639 [============================>.] - ETA: 0s - loss: 0.0427 - acc: 0.8530^@^@^@^@^@^@^@^@
Epoch 00002: val_acc did not improve from 0.88778
1639/1639 [==============================] - 1048s 640ms/step - loss: 0.0427 - acc: 0.8550 - val_loss: 0.0585 - val_acc: 0.8169

Epoch 00003: LearningRateScheduler reducing learning rate to 0.0056666667473812896.
Epoch 3/100
1638/1639 [============================>.] - ETA: 0s - loss: 0.0382 - acc: 0.8677^@^@^@^@^@^@^@
Epoch 00003: val_acc did not improve from 0.88778
1639/1639 [==============================] - 1044s 637ms/step - loss: 0.0382 - acc: 0.8661 - val_loss: 0.0480 - val_acc: 0.8352

Epoch 00004: LearningRateScheduler reducing learning rate to 0.007999999997516473.
Epoch 4/100
1638/1639 [============================>.] - ETA: 0s - loss: 0.0364 - acc: 0.8727^@^@^@^@^@^@^@
Epoch 00004: val_acc did not improve from 0.88778
1639/1639 [==============================] - 1038s 633ms/step - loss: 0.0364 - acc: 0.8727 - val_loss: 0.5048 - val_acc: 0.1988

Epoch 00005: LearningRateScheduler reducing learning rate to 0.00800000037997961.
```

**Comment by Saduf2019** [comment]
@apatsekin 
We see that you are using old version of tensorflow which is not actively supported now, please verify on 2.x and open a new issue in case you face any errors, hence moving this to closed status.
